### ISSUE TITLE
Data format for image and labels

### ISSUE THREAD

**Issue by mbellitti** [mbellitti/wikiart-classifier#1]
We have to choose how to save the images downloaded by the scraper in a way that will be easy to handle with Keras or `sklearn` later.

As @janpfsr suggested, we could look at how the CIFAR solved this problem. I think this is a very good starting point.

The `pillow` library should have the functionality to label the images: the do.cs are [here](https://pillow.readthedocs.io/en/5.1.x/)

**Comment by mbellitti** [comment]
It looks like the best way to do this is by using a pandas dataframe, since Keras has a nice interface towards that format. 

Every entry should be the metadata collected with the `image_scraper` library, plus a column that contains a filename.

In this way we have both access to the image url and a file on disk (if it exists), which makes the thing more flexible. We should then estimate what the size of the database would be if we download all the artworks, and decide if and where we want to store those. Maybe ask Pankaj to give us some extra space on the cluster?

**Comment by mohitpandey92** [comment]
I have written code for this on the branch "df_pandas".

**Comment by mbellitti** [comment]
Great! Once everything works and it's documented merge it into `master` and delete the `df_pandas` branch.
Also, please keep all the source code inside the `src/` directory.

**Comment by mohitpandey92** [comment]
As of now, the last column is file_name. I suppose once I have finalized the code, we should pre-process the data and the last column of JSON file should store pixel values of pre-processed image.

**Comment by mbellitti** [comment]
I would prefer keeping it the way it is now, otherwise the database will become memory-heavy. 
A possibility would be to create a method that loads the image in memory only when it needs it, and then unloads it once the calculation is done.

**Comment by mohitpandey92** [comment]
I have merged the branch "df_pandas" with the master.

**Comment by mbellitti** [comment]
The images will be kept in the /data/images/ directory, the column "_id" contains the filename of corresponding image.
